### ISSUE TITLE
[consistent-indexed-object-style] Fix produces invalid recursive type alias

### ISSUE THREAD

**Issue by andersk** [typescript-eslint/typescript-eslint#2787]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/consistent-indexed-object-style": "error"
  }
}
```

```TS
type Tree = { [key: string]: Tree } | null;
```

**Expected Result**

No error—this recursive indexed object type would become invalid if converted to `Record`.

**Actual Result**

```
$ tsc src.ts
$ eslint src.ts

/tmp/test/src.ts
  1:13  error  A record is preferred over an index signature  @typescript-eslint/consistent-indexed-object-style

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Fixing this error with `--fix` produces an invalid recursive type alias:

```TS
type Tree = Record<string, Tree> | null;
```

```
$ tsc src.ts
src.ts:1:6 - error TS2456: Type alias 'Tree' circularly references itself.

1 type Tree = Record<string, Tree> | null;
       ~~~~


Found 1 error.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.8.1` |
| `@typescript-eslint/parser`        | `4.8.1` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.13.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2687

Please use the search

**Comment by andersk** [comment]
Oh yeah, sorry. (I was searching for “recursive type alias”, which is what the TypeScript documentation calls it, but certainly I could have searched harder.)
